benches/bench_flows.py

```diff
@@ -55,7 +55,10 @@ def benchmark_flow():
         for _ in range(num_tasks):
             test_task()
 
-    benchmark(benchmark_flow)
+    if num_tasks > 100:
+        benchmark.pedantic(benchmark_flow)
+    else:
+        benchmark(benchmark_flow)
```

Comment:
I was seeing the benchmarks timeout because `benchmark` run the wrapped function multiple times to get a enough data points to determine an accurate runtime. When there's more than 100 task it takes too long right now. `benchmark.pedantic` ensures that we only run the flow once.

Previously, the benchmarks were unintentionally running against the ephemeral API. I fixed that, and now they run faster. Funnily enough, I think that led to timeouts because `pytest-benchmark` was running more iterations.